### PR TITLE
feat(debug): Add llb export

### PR DESCRIPTION
**Pull request by gaocegege**
Fix https://github.com/tensorchord/envd/issues/896

```
./bin/envd --debug debug llb -p ./examples/python-basic --dot > a.dot
dot -Tpng a.dot -o a.png
```

a

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

File: pkg/app/debug_llb.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+		}
+	}
+}
+
+func attr(dgst digest.Digest, op pb.Op) (string, string) {
+	switch op := op.Op.(type) {
+	case *pb.Op_Source:
+		return op.Source.Identifier, "ellipse"
+	case *pb.Op_Exec:
+		return strings.Join(op.Exec.Meta.Args, " "), "box"
```
Comment (VoVAllen) [inline pkg/app/debug_llb.go:168]: Can also include mount information with `op.Exec.Mounts`

**Review by VoVAllen — DISMISSED** [review]
Just realized the cache key is generated at runtime instead of graph build time 😢  Good feature but seems won't solve the cache debugging problem

**Comment by gaocegege** [comment]
> Just realized the cache key is generated at runtime instead of graph build time cry Good feature but seems won't solve the cache debugging problem

The cache is calculated from the build time info, thus you can use the graph to simulate the process.

Ref https://github.com/moby/buildkit/issues/2383

**Comment by aseaday** [comment]
LGTM. To better solve debug problem. Could we have time record on the graph?

**Comment by gaocegege** [comment]
> etter solve debug problem. Could we have time record on the graph?

It is not possible to do that because the graph is static, and the build is actually not executed.

**Pull request merged**